lib/_stream_wrap.js

```diff
@@ -137,10 +137,8 @@ StreamWrap.prototype.doWrite = function doWrite(req, bufs) {
 
       var errCode = 0;
       if (err) {
-        if (err.code && uv['UV_' + err.code])
-          errCode = uv['UV_' + err.code];
-        else
-          errCode = uv.UV_EPIPE;
+        const code = uv[`UV_${err.code}`];
+        errCode = err.code && code ? code : uv.UV_EPIPE;
```

Comment:
Suggestions:
* This closes over `err`, why not hoist it out of the callback, so only `errCode` gets closed.
* if `!err.code` then `` `UV_${err.code}` == somethinglike('UV_undefined')`` so `` uv[`UV_${err.code}`] == false``
* IIFE can be nice:
```js
const errCode = (() => {
  if (!err) return 0;
  return uv[`UV_${err.code}`] || uv.UV_EPIPE;
})();